packages/dx-grid-core/src/plugins/sorting-state/helpers.js

```diff
@@ -15,11 +10,25 @@ export const getPersistentSortedColumns = (sorting, columnExtensions = []) =>
     return acc;
   }, []);
 
-export const calculateKeepOther = (sorting, keepOther, persistentSortedColumns = []) => {
+const calculateKeepOther = (sorting, keepOther, columnExtensions) => {
+  const persistentSortedColumns =
+    getPersistentSortedColumns(sorting, columnExtensions);
   if (!persistentSortedColumns.length) return keepOther;
   if (!keepOther) return persistentSortedColumns;
 
   return Array.isArray(keepOther)
     ? unique([...keepOther, ...persistentSortedColumns])
     : unique([...sorting.map(item => item.columnName), ...persistentSortedColumns]);
 };
+
+export const getColumnSortingDirection = (sorting, columnName) => {
+  const columnSorting = sorting.filter(s => s.columnName === columnName)[0];
+  return columnSorting ? columnSorting.direction : null;
+};
+
+export const getChangeColumnSorting =
```

Comment:
Can we somehow avoid a function returning from a function?

Reply:
Well, `getChangeColumnSorting` is a function that returns a reducer. Like decorator.